File: app/src/main/java/com/examle/curexchange/ui/second/SecondCurrencyPresenter.java
package com.examle.curexchange.ui.second;

import com.arellomobile.mvp.InjectViewState;
import com.examle.curexchange.ui.base.BasePresenter;

import java.util.List;

@InjectViewState
public class SecondCurrencyPresenter extends BasePresenter<SecondCurrencyView> {

    private List<String> names;
    private String firstName;
    private int value;

    public SecondCurrencyPresenter(List<String> names, String firstName, int value) {
        this.names = names;
        this.firstName = firstName;
        this.value = value;
    }

    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
        getViewState().showData(names);
    }

    void buttonClicked() {
        getViewState().handleFloatButton(firstName, value);
    }
}

File: app/src/main/java/com/examle/curexchange/ui/second/SecondCurrencyPresenter.kt
package com.examle.curexchange.ui.second

import com.arellomobile.mvp.InjectViewState
import com.examle.curexchange.ui.base.BasePresenter

@InjectViewState
class SecondCurrencyPresenter(var names: List<String>, var firstName: String, var value: Int) :
        BasePresenter<SecondCurrencyView>() {

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()
        viewState.showData(names)
    }

    fun onFloatButtonExchangeClicked() {
        viewState.handleFloatButton(firstName, value)
    }
}